vpr/src/place/place_constraints.cpp

```diff
@@ -54,8 +54,8 @@ bool is_macro_constrained(const t_pl_macro& pl_macro) {
 }
 
 /*Returns PartitionRegion of where the head of the macro could go*/
-PartitionRegion constrained_macro_locs(const t_pl_macro& pl_macro) {
-    PartitionRegion macro_pr;
+PartitionRegion update_macro_head_pr(const t_pl_macro& pl_macro) {
```

Comment:
Should update the partition regions on all macro members, not just the head.  Easy to do: after you've found the implied (tightest) constraint on the macro head, just shift it the appropriate amount for each other member of the macro (i.e. if the next member is at delta_y = +1, then shift the y-values by 1),. Also should intersect the resulting constraint per member with the grid dimensions, unless you can prove that the code that figures out the constraint on the macro_head guarantees that later shifting it can never go outside the grid.

Reply:
Updating the PartitionRegions of the other members is done with the helper function update macro_member_pr. Yes, I can update that function to intersect with the grid dimensions. I'll also add a check to see whether the updated PartRegion is empty, just like I have in the head macro function, because that would indicate that the floorplan constraints are incompatible as well.